### ISSUE TITLE
DetailsList not announcing new state of items, when they change because of filtering

### ISSUE THREAD

**Issue by nkalpana** [OfficeDev/office-ui-fabric-react#11281]
### Environment Information

- **Package version(s)**: 7.31
- **Browser and OS versions**:  Edge + Narrator, Chrome + NVDA, Windows 10

### Describe the issue:
Filtered suggestions, or "no results found" not announced when user types in the filter string in the filter text box for DetailsList control

### Please provide a reproduction of the issue in a codepen:
https://codepen.io/nkalpana/pen/zYYXWqX

Open in Edge/Chrome and use Narrator/NVDA as your screenreader.
Type into filter text box, with strings matching the contents of the table like "Lore", and also try a string that does not match with any entry. 
The filtered new items of the table, or how many new items are there, or if table is empty because of no match, are not announced by screenreader.

#### Actual behavior:

Only the filter string as typed into the text box is announced. The new changed items list or empty table because of no match are not announced by screenreader

#### Expected behavior:

Only the filter string as typed into the text box is announced. The new changed items list or empty table because of no match are not announced by screenreader

**Comment by aneeshack4** [comment]
@nkalpana Thanks for reporting this. It does repro and it doesn't announce the results of filtering, whether items were found matching the string or not. I believe this is because in this filtered example, the `TextField` and the `MarqueeSelection` with the `DetailsList` inside it aren't really connected - or one whole unit, they're acting sort of independently. The _onFilter method triggered by the `TextField` `onChange` simply modifies the state of the `items` of the `DetailsList` but nothing else. So maybe this `_onFilter` method needs to change the `ariaLabel` value of the `TextField` and `DetailsList` to trigger Narrator announcing the changes. I'll speak with the component owners and follow up.

**Comment by nkalpana** [comment]
Yes, the filter and the details list are independent issues. However the testers were expecting the DetailsList to react to the change of "items" property and say that 14 rows.

Please do let me know after your discussion. Thanks for looking into this!

**Comment by aneeshack4** [comment]
@nkalpana The DetailsList component needs a lot of additional work and we're working on finding a component owner for it/figuring out its future so I'll add this to the backlog for review when we start doing that.

**Comment by khmakoto** [comment]
Hi @nkalpana, a similar issue came before and we've exemplified in all of our examples with a filter how to achieve this with the use of the `Announced` component. We've decided against baking this into the control since the "filtering" is being done by the user and it's not baked in itself. Please let us know if we can help you in any other way!
